src/pages/preview.jsx

```diff
@@ -0,0 +1,60 @@
+import React, { useEffect } from 'react';
+
+import PropTypes from 'prop-types';
+
+const GITHUB_API_BASE_URL =
+  'https://api.github.com/repos/newrelic/newrelic-quickstarts/contents';
+
+const GITHUB_API_PULL_URL =
+  'https://api.github.com/repos/newrelic/newrelic-quickstarts/pulls';
+
+/**
+ * Recursive function to walk the file system in GitHub
+ * @param {String} url - GitHub API URL to walk the file system
+ * @returns {Array} array of files
+ **/
+const iterateDirs = async (files, url) => {
+  const branchResponse = await fetch(url);
+  const branchResponseJSON = await branchResponse.json();
+
+  for (const dirListing of branchResponseJSON) {
+    if (dirListing.type === 'file') {
+      files.push(dirListing);
+    } else {
+      await iterateDirs(files, dirListing.url);
+    }
+  }
+};
+
+const fetchData = async (prNumber, quickstartPath) => {
```

Comment:
I think giving this a more descriptive name would help out, maybe something like
```suggestion
const getQuickstartFilesFromPR = async (prNumber, quickstartPath) => {
```